pkg/promql/engine.go

```diff
@@ -743,22 +728,17 @@ func (ng *Engine) populateSeries(ctx context.Context, querier Querier, s *parser
 			}
 
 			if topNode != nil {
-				return fmt.Errorf("Assigning topNode twice")
-			}
-			set, topNode, wrn, err = querier.Select(false, hints, path, n.LabelMatchers...)
-			warnings = append(warnings, wrn...)
-			if err != nil {
-				level.Error(ng.logger).Log("msg", "error selecting series set", "err", err)
-				return err
+				return fmt.Errorf("assigning topNode twice")
 			}
+			set, topNode = querier.Select(false, hints, path, n.LabelMatchers...)
```

Comment:
hmm (for another PR) should we consider making topNode also part of the SeriesSet?

Reply:
We anyhow need to keep topNode in evaluator for checks in `ev.eval()`. (I think so, not sure).

Reply:
Sure, will open an issue for this.